### PR TITLE
[docs] redesign: switch to new Terminal component

### PR DESCRIPTION
**Pull request by Simek**
# Why

This PR switches the all Terminal blocks already used in the docs to the new component. This is a part of docs redesign.

# How

`TerminalBlock` has been replaced with `Terminal` from new Snippets. Since the new component supports command copy, I had to add the copy friendly string in `cmdCopy` prop.

There was an issue, when lines without the prefix characters were incorrectly rendered (inline with previous lines), but I was able to fix this, but altering the `Terminal` styles a bit.

In this changeset I have also correct few links and clean up few HTML/JSX tags on the edited pages.

# Test Plan

The changes have been tested by running docs website locally.

# Preview

Screenshot 2022-04-19 at 13 07 43
Screenshot 2022-04-19 at 13 07 52

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/background-fetch.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-background-
 packageName: 'expo-background-fetch'
 ---
 
-import APISection from '~/components/plugins/APISection';
+import {APIInstallSection} from '~/components/plugins/InstallSection';
```
Comment (FiberJW) [inline docs/pages/versions/unversioned/sdk/background-fetch.md:7]:
```suggestion
import { APIInstallSection } from '~/components/plugins/InstallSection';
```

**Review by FiberJW — APPROVED** [review]
Nice! I found 2 spots that could use a formatting fix before merging

One thing that I notice about this API is that it seems to duplicate a lot of text, which makes it easier for the terminal display and the command you copy to go out of sync.

Maybe the Terminal component could build a copyable string with tagged React.Children?

Or maybe the array we pass into `cmd` could be list of 'tokens' that you can build the layout with and extract the copyable content from (`{ content: string; endOfLine: boolean; copyable: boolean }`). Just an idea though!

**Review by tsapeta — COMMENTED** [review]
In the future we may want to use this component also on the modules API page, like here: https://docs.expo.dev/modules/module-api/ (look at the side-by-side examples for Swift and Kotlin). Do you think it'll be feasible?

**Review by brentvatne — APPROVED** [review]
:shipit:

**Comment by Simek** [comment]
> One thing that I notice about this API is that it seems to duplicate a lot of text, which makes it easier for the terminal display and the command you copy to go out of sync.

Yeah, I agree, I have spotted that during the refactor, that adding manually copy command is bit repetitive and probably can be automated based on the content. However here I was focused on the replacement of blocks, and I have used the `TerminalBlock` component developed for Home page. I will look into the automation idea and will push separate PR improving that in the near future. 🙂 

> In the future we may want to use this component also on the modules API page, like here: https://docs.expo.dev/modules/module-api/ (look at the side-by-side examples for Swift and Kotlin). Do you think it'll be feasible?

Thanks for mentioning that! 👍 There is the the new `CodeBlock` design ready, it will be based on the `Snippet` component, so the title and copy functionalities will there. We can add special version or a wrapper which enable to display two block side by side. We also will have that in mind during implementation.

**Pull request merged**